src/solver/libModelObject/include/antares/solver/libObjectModel/library.h

```diff
@@ -32,15 +33,56 @@ namespace Antares::Solver::ObjectModel
 class Library
 {
 public:
-    Library();
+    Library() = default;
     ~Library() = default;
 
+    const std::string& id() const
+    {
+        return id_;
+    }
+
+    const std::string& description() const
+    {
+        return description_;
+    }
+
+    const std::map<std::string, PortType>& portTypes() const
+    {
+        return portTypes_;
+    }
+
+    const std::map<std::string, Model>& models() const
+    {
+        return models_;
+    }
+
 private:
+    friend class LibraryBuilder;
+
     std::string id_;
     std::string description_;
 
     std::map<std::string, PortType> portTypes_;
     std::map<std::string, Model> models_;
 };
 
+/**
+ * @brief Builder for the Library class
+ * Follow builder pattern:
+ * builder.Library().withId("id").withDescription("description").withPortType(portList).withModel(modelList).build();
+ */
+class LibraryBuilder
+{
+public:
+    LibraryBuilder& withId(const std::string& id);
+    LibraryBuilder& withDescription(const std::string& description);
+    LibraryBuilder& withPortType(std::vector<PortType>& portTypes);
+    LibraryBuilder& withModel(std::vector<Model>& models);
```

Comment:
if you want to go further : instead of adding a built model, you can declare a "newModel()" method that returns a ModelBuilder

Reply:
Same as https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2431#discussion_r1784571025